proto/ibc/core/channel/v1/upgrade.proto

```diff
@@ -6,6 +6,25 @@ option go_package = "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types";
 
 import "gogoproto/gogo.proto";
 import "ibc/core/client/v1/client.proto";
+import "ibc/core/channel/v1/channel.proto";
+
+// Upgrade is a verifiable type which contains the relevant information
+// for an attempted upgrade. It provides the proposed changes to the channel
+// end, the timeout for this upgrade attempt and the last packet sequence sent
+// to allow the counterparty to block sends after the upgrade has started.
+message Upgrade {
+  ModifiableUpgradeFields upgrade_fields   = 1 [(gogoproto.nullable) = false];
+  UpgradeTimeout          timeout          = 2 [(gogoproto.nullable) = false];
+  uint64                  last_packet_sent = 3;
+}
+
+// ModifiableUpgradeFields are the fields in a channel end which may be changed
+// during a channel upgrade.
+message ModifiableUpgradeFields {
```

Comment:
what do you think about calling this just `UpgradeFields`?

Reply:
yeah I think either name could works, the `Modifiable` very explicit, but I don't feel super strongly about it.